### ISSUE TITLE
Jenkins doesn't have label XYZ

### ISSUE THREAD

**Issue by ppitonak** [openshiftio/openshift.io#4744]
##### Issue Overview

Pipeline gets stuck at the beginning and Jenkins log only shows message `'Jenkins' doesn't have label 'mavenPROJECTNAME_1'` and a spinner

##### Expected Behaviour

1. Jenkins slave is created and pipeline finishes successfully
2. The message explains the situation more clearly

##### Current Behaviour

![crt_jenkins_label](https://user-images.githubusercontent.com/445100/51978167-ad697580-2489-11e9-8ee0-050ba5551395.png)

##### Steps To Reproduce

1. create new space
2. create new SpringBoot app
3. open pipeline log

##### Additional Information

Observable in this job https://ci.centos.org/job/devtools-test-e2e-openshift.io-smoketest-us-east-2-released/1506/console

The broken functionality is a regression, originally solved in https://github.com/openshiftio/openshift.io/issues/4020

The UX part was supposed to be fixed by https://github.com/openshiftio/openshift.io/issues/3797

**Comment by chmouel** [comment]
> The broken functionality is a regression, originally solved in #4020

I don't believe this is solved in 4020, 

"Jenkins doesn't have label" is a catch all sentence, which in english would mean, "i have asked to get a new pod with this label but it didn't came up"

I can't get a pod can mean a lot of things,  99% of the time is beause of an infrastrcture issue, without logs it's hard to say,

**Comment by piyush-garg** [comment]
@ppitonak The changed message appears when slave pod gets ready, but till that time it does not print anything. In our case, the slave pod was not provisioned that's why we did not see the message, otherwise, if you see any other job this message is still there.

**Comment by jarifibrahim** [comment]
This could be related to https://github.com/openshiftio/openshift.io/issues/4745. https://github.com/openshiftio/openshift.io/issues/4745 was also seen on cluster 2.

**Comment by Preeticp** [comment]
@jarifibrahim you are right, as mentioned in the last screenshot for #4745 I get a similar error message in my second attempt and the pipeline seems to be stuck forever (19 hours now).
